compiler/rustc_parse/src/parser/stmt.rs

```diff
@@ -555,9 +555,15 @@ impl<'a> Parser<'a> {
             StmtKind::Expr(ref mut expr)
                 if self.token != token::Eof && classify::expr_requires_semi_to_be_stmt(expr) =>
             {
+
+                let res = self.expect_one_of(&[], &[token::Semi, token::CloseDelim(Delimiter::Brace)]);
+                let mut skip_type_error = false;
+                if let Ok(true) = &res {
```

Comment:
Please use a match here instead of assigning `res` to a local variable.

Reply:
Also, can you comment _why_ this works? It's not clear from the usage that `Ok(true)` means "recovered".